### PR TITLE
Add a simply JSON tidier that preserves map key order

### PR DESCRIPTION
**Pull request by autarch**

### CODE REVIEW

File: cmd/json-ordered-tidy/main.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	f, err := os.Open(dir)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not open directory %s: %s\n", dir, err)
+		return
+	}
+
+	names, err := f.Readdirnames(0)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not read directory contents of %s: %s\n", dir, err)
+	}
```
Comment (smreichling) [inline cmd/json-ordered-tidy/main.go]: missing return

File: cmd/json-ordered-tidy/main.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+func (t *tidier) handlePath(path string) {
+	fi, err := os.Stat(path)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error stat'ing path %s: %s\n", path, err)
+		return
+	}
+	if fi.IsDir() {
+		t.recurseDir(path)
+	} else if regexp.MustCompile(regexp.QuoteMeta(t.ext) + `$`).MatchString(path) {
```
Comment (smreichling) [inline cmd/json-ordered-tidy/main.go]: I might move this regex into the `tidier` struct and compile it once in the main method so it's not be recompiled per-file.

**Review by smreichling — CHANGES_REQUESTED** [review]

**Comment by autarch** [comment]
Addressed comments and rebased.

**Comment by autarch** [comment]
@smreichling It's ready for you to review again.

**Review by smreichling — APPROVED** [review]

**Pull request merged**